appengine-managed-runtime/src/main/java/com/google/apphosting/vmruntime/VmRuntimeFileLogHandler.java

```diff
@@ -102,4 +111,57 @@ private static void reloadLoggingProperties(LogManager logManager) {
       System.err.println(e.getClass().getName() + ": " + e.getMessage());
     }
   }
+
+  @Override
+  public void close() throws SecurityException {
+    super.close();
+    if (handlers.decrementAndGet() <= 0 ) {
+      queue.offer(null);
+    }
+  }
+
+  @Override
+  public void publish(LogRecord record) {
+    if (isLoggable(record)) {
+      queue.offer(new LogEntry(record));
```

Comment:
Currently I have gone with an unlimited log queue.   If you look at the Juli example linked to, you will see that they have a fixed sized queue with configurable actions to do if the queue is full (discard, block etc.).

My thoughts on this is that any app that is suffering OOM issues because of a large logevent queue is probably going to have similar issues anyway, even if the logevents are discarded.     I think an unlimited queue is thus the simplest and does not need configuration and/or tuning.

So if the unlimited policy is accepted, the offer return does not need to be checked.   However I'm open to be convinced that another policy may be better?